haystack_experimental/components/extractors/llm_document_content_extractor.py

```diff
@@ -0,0 +1,248 @@
+            when working with models that have resolution constraints or when transmitting images to remote services.
+        :param raise_on_failure: If True, exceptions from the LLM are raised. If False, failed documents are logged
+            and returned.
+        :param max_workers: Maximum number of threads used to parallelize LLM calls across documents using a
+            ThreadPoolExecutor.
+        """
+        # Needed for DocumentToImageContent component
+        pillow_import.check()
+        pypdfium2_import.check()
+
+        self._chat_generator = chat_generator
+        self.prompt = prompt
+        self.file_path_meta_field = file_path_meta_field
+        self.root_path = root_path or ""
+        self.detail = detail
+        self.size = size
+        # Ensure the prompt does not contain any variables.
+        ast = SandboxedEnvironment().parse(prompt)
+        template_variables = meta.find_undeclared_variables(ast)
+        variables = list(template_variables)
+        if len(variables) != 0:
+            raise ValueError(
+                f"The prompt must not have any variables only instructions on how to extract the content of the "
+                f"image-based document. Found {','.join(variables)} in the prompt."
+            )
+        self.raise_on_failure = raise_on_failure
+        self.max_workers = max_workers
+        self._document_to_image_content = DocumentToImageContent(
+            file_path_meta_field=file_path_meta_field,
+            root_path=root_path,
+            detail=detail,
+            size=size,
+        )
+
+    def warm_up(self):
+        """
+        Warm up the ChatGenerator if it has a warm_up method.
+        """
+        if hasattr(self._chat_generator, "warm_up"):
+            self._chat_generator.warm_up()
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serializes the component to a dictionary.
+
+        :returns:
+            Dictionary with serialized data.
+        """
+
+        return default_to_dict(
+            self,
+            chat_generator=component_to_dict(obj=self._chat_generator, name="chat_generator"),
+            prompt=self.prompt,
+            file_path_meta_field=self.file_path_meta_field,
+            root_path=self.root_path,
+            detail=self.detail,
+            size=self.size,
+            raise_on_failure=self.raise_on_failure,
+            max_workers=self.max_workers,
+        )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "LLMDocumentContentExtractor":
+        """
+        Deserializes the component from a dictionary.
+
+        :param data:
+            Dictionary with serialized data.
+        :returns:
+            An instance of the component.
+        """
+        deserialize_chatgenerator_inplace(data["init_parameters"], key="chat_generator")
+        return default_from_dict(cls, data)
+
+    def _run_on_thread(self, prompt: Optional[ChatMessage]) -> Dict[str, Any]:
```

Comment:
nit: for clarity, I would rename `prompt` to `message` or something similar. I would do that here and in `run`